dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,288 @@
+    auto full_arg = make_span(full.data() + sample_idx * spatial_ndim, spatial_ndim);
+    for (int i = 0; i < nargs; i++) {
+      int d = axes[i] - first_spatial_dim;
+      assert(d >= 0);
+      assert(d < spatial_ndim);
+      full_arg[d] = orig_arg[i];
+    }
+  }
+}
+
+void TensorResizeAttr::PrepareResizeParams(const OpSpec &spec, const ArgumentWorkspace &ws,
+                                           const TensorListShape<> &input_shape) {
+  ndim_ = input_shape.sample_dim();
+  if (ndim_ < 1)
+    throw std::invalid_argument("Expected at least 1D inputs");
+  if (!has_axes_) {
+    axes_.resize(ndim_);
+    std::iota(axes_.begin(), axes_.end(), 0);
+  }
+  // TODO(janton): use utils
+  kernels::reduce_impl::CheckAxes(make_cspan(axes_), ndim_);
+  kernels::reduce_impl::AdjustAxes(make_span(axes_), ndim_);
+
+  int nargs = axes_.size();
+  int first_axis = ndim_;
+  int last_axis = -1;
+  for (int axis : axes_) {
+    if (axis < first_axis)
+      first_axis = axis;
+    if (axis > last_axis)
+      last_axis = axis;
+  }
+  spatial_ndim_ = last_axis - first_axis + 1;
+  first_spatial_dim_ = first_axis;
+
+  int nsamples = input_shape.num_samples();
+  params_.resize(nsamples);
+
+  auto axes = make_cspan(axes_);
+
+  if (has_alignment_) {
+    GetShapeLikeWithAxes<float>(alignment_, alignment_arg_, spec, ws, "alignment", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+  }
+
+  if (has_sizes_) {
+    GetShapeLikeWithAxes<float>(sizes_, sizes_arg_, spec, ws, "sizes", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+  } else if (has_scales_) {
+    GetShapeLikeWithAxes<float>(scales_, scales_arg_, spec, ws, "scales", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+  } else {
+    assert(false);  // should not happen
+  }
+
+  if (has_roi_) {
+    GetShapeLikeWithAxes<float>(roi_start_, roi_start_arg_, spec, ws, "roi_start", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+    GetShapeLikeWithAxes<float>(roi_end_, roi_end_arg_, spec, ws, "roi_end", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+  }
+
+  max_size_.resize(spatial_ndim_,
+                   std::nextafter(static_cast<float>(std::numeric_limits<int>::max()), 0.0f));
+  if (has_max_size_) {
+    GetSingleOrRepeatedArg(spec, max_size_arg_, "max_size", nargs);
+    for (int i = 0; i < nargs; i++) {
+      int d = axes_[i] - first_spatial_dim_;
+      assert(d >= 0);
+      assert(d < spatial_ndim_);
+      max_size_[d] = max_size_arg_[i];
+    }
+  }
+
+  SmallVector<float, 3> requested_size, in_lo, in_hi;
+  requested_size.resize(spatial_ndim_);
+  in_lo.resize(spatial_ndim_);
+  in_hi.resize(spatial_ndim_);
+
+  assert(has_sizes_ + has_scales_ == 1);
+  const float *max_size = has_max_size_ ? max_size_.data() : nullptr;
```

Comment:
Maybe it is nitpicky, but could you move `max_size = nullptr` declaration to L204? There is the section that handles this param and then it is only used in `CalculateSampleParams`. Here, it is kind of interleaved with `requested_size, in_lo, in_hi`.